pkg/commands/resolver.go

```diff
@@ -122,6 +122,14 @@ func gobuildOptions(bo *options.BuildOptions) ([]build.Option, error) {
 		opts = append(opts, build.WithConfig(bo.BuildConfigs))
 	}
 
+	if bo.ImageConfigs != nil {
+		opts = append(opts, build.WithImageConfig(bo.ImageConfigs))
```

Comment:
What if `WithImageConfig` just handled `nil`? Then we could `opts = append(opts, build.WithImageConfig(bo.ImageConfigs))` no matter what. (same with `defaultImageConfig`)